app/scripts/models/auth_brokers/oauth.js

```diff
@@ -81,17 +82,95 @@ define(function (require, exports, module) {
       if (! account || ! account.get('sessionToken')) {
         return p.reject(AuthErrors.toError('INVALID_TOKEN'));
       }
-
+      function ua2hex(uint8arr) {
+        if (!uint8arr) {
+          return '';
+        }
+
+        var hexStr = '';
+        for (var i = 0; i < uint8arr.length; i++) {
+          var hex = (uint8arr[i] & 0xff).toString(16);
+          hex = (hex.length === 1) ? '0' + hex : hex;
+          hexStr += hex;
+        }
+
+        return hexStr.toUpperCase();
+      }
+      var asser;
+      var keys;
       const relier = this.relier;
       const clientId = relier.get('clientId');
       return this._assertionLibrary.generate(account.get('sessionToken'), null, clientId)
         .then((assertion) => {
+          asser = assertion
+          var keyFetchToken = account.get('keyFetchToken');
+          var unwrapBKey = account.get('unwrapBKey');
+
+          console.log({
+            keyFetchToken: keyFetchToken,
+            unwrapBKey: unwrapBKey
+          });
+
+          return this._fxaClient.accountKeys(keyFetchToken, unwrapBKey)
+
+
+        })
+        .then((rkeys) => {
+          var uid = account.get('uid');
+          return this.relier.deriveRelierKeys(rkeys, uid);
+        })
+        .then((rkeys) => {
+          keys = rkeys;
+          var pk = JSON.parse(relier.get('jwk'))
+
+          return window.crypto.subtle.importKey(
```

Comment:
Nah, I think we are going for Firefox 57 for now (in the next iteration Android 4+)